tensornetwork/linalg/linalg.py

```diff
@@ -0,0 +1,191 @@
+from tensornetwork import network_components
+
+
+Tensor = Any
+BaseBackend = base_backend.BaseBackend
+
+
+# INITIALIZATION
+def initialize_node(fname: Text,
+                    *fargs: Any,
+                    name: Optional[Text] = None,
+                    axis_names: Optional[List[Text]] = None,
+                    backend: Optional[Union[Text, BaseBackend]] = None,
+                    **fkwargs: Any):
+  """Return a Node wrapping data obtained by an initialization function
+  implemented in a backend. The Node will have the same shape as the
+  underlying array that function generates, with all Edges dangling.
+
+  This function is not intended to be called directly, but doing so should
+  be safe enough.
+  Args:
+    fname:  Name of the method of backend to call (a string).
+    *fargs: Positional arguments to the  initialization method.
+    name: Optional name of the Node.
+    axis_names: Optional names of the Node's dangling edges.
+    backend: The backend or its name.
+    **fkwargs: Keyword arguments to the initialization method.
+
+  Returns:
+    node: A Node wrapping data generated by
+          (the_backend).fname(*fargs, **kwargs), with one dangling edge per
+          axis of data.
+  """
+  if backend is None:
+    backend_obj = backend_contextmanager.get_default_backend()
+  else:
+    backend_obj = backends.backend_factory.get_backend(backend)
+  func = getattr(backend_obj, fname)
+  data = func(*fargs, **fkwargs)
+  node = network_components.Node(data, name=name, axis_names=axis_names,
+                                 backend=backend)
+  return node
+
+
+def eye(N: int,
+        dtype: Optional[Type[np.number]],
+        M: Optional[int] = None,
+        name: Optional[Text] = None,
+        axis_names: Optional[List[Text]] = None,
+        backend: Optional[Union[Text, BaseBackend]] = None) -> Tensor:
+  """Return a Node representing a 2D array with ones on the diagonal and
+  zeros elsewhere. The Node has two dangling Edges.
+  Args:
+    N (int): The first dimension of the returned matrix.
+    dtype, optional: The dtype of the returned matrix.
+    M (int, optional): The second dimension of the returned matrix.
+    name (text, optional): Name of the Node.
+    axis_names (optional): List of names of the edges.
+    backend (optional): The backend or its name.
+
+  Returns:
+    I : Node of shape (N, M)
+        Represents an array of all zeros except for the k'th diagonal of all
+        ones.
+  """
+  the_node = initialize_node("eye", N,
+                             name=name, axis_names=axis_names, backend=backend,
+                             dtype=dtype, M=M)
+  return the_node
+
+
+def zeros(shape: Sequence[int],
+          dtype: Optional[Type[np.number]],
+          name: Optional[Text] = None,
+          axis_names: Optional[List[Text]] = None,
+          backend: Optional[Union[Text, BaseBackend]] = None) -> Tensor:
+  """Return a Node of shape 'shape' of all zeros.
```

Comment:
use `shape` instead of 'shape'

Reply:
Do you mean the word shape or the Markdown-formatted `shape`